packages/x-data-grid-pro/src/hooks/features/dataSource/useGridDataSourceCache.ts

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import useLazyRef from '@mui/utils/useLazyRef';
+import { useGridApiMethod } from '@mui/x-data-grid';
+import { GridPrivateApiPro } from '../../../models/gridApiPro';
+import { DataGridProProcessedProps } from '../../../models/dataGridProProps';
+import { GridGetRowsParams, GridGetRowsResponse, GridDataSourceCache } from '../../../models';
+import { GridDataSourceCacheApi } from './interfaces';
+import { SimpleServerSideCache } from './cache';
+
+const getDefaultCache = (cacheInstance: SimpleServerSideCache): GridDataSourceCache => ({
+  getKey: cacheInstance.getKey,
+  set: (key: string, value: GridGetRowsResponse) =>
+    cacheInstance.set(key as string, value as GridGetRowsResponse),
+  get: (key: string) => cacheInstance.get(key as string),
+  clear: () => cacheInstance.clear(),
+});
+
+export const useGridDataSourceCache = (
+  privateApiRef: React.MutableRefObject<GridPrivateApiPro>,
+  props: Pick<DataGridProProcessedProps, 'unstable_dataSource' | 'unstable_dataSourceCache'>,
+): void => {
+  const defaultCache = useLazyRef<GridDataSourceCache, void>(() =>
+    getDefaultCache(new SimpleServerSideCache({})),
+  );
+  const cache = React.useRef<GridDataSourceCache | null>(
+    props.unstable_dataSourceCache || defaultCache.current,
+  );
+
+  const getCacheData = React.useCallback((params: GridGetRowsParams) => {
+    if (!cache.current) {
+      return undefined;
+    }
+    const key = cache.current.getKey(params);
+    return cache.current.get(key);
+  }, []);
+
+  const setCacheData = React.useCallback((params: GridGetRowsParams, data: GridGetRowsResponse) => {
+    if (!cache.current) {
+      return;
+    }
+    const key = cache.current.getKey(params);
+    cache.current.set(key, data);
+  }, []);
+
+  const clearCache = React.useCallback(() => {
+    if (!cache.current) {
+      return;
+    }
+    cache.current.clear();
+  }, []);
+
+  const dataSourceCacheApi: GridDataSourceCacheApi = {
```

Comment:
How about exposing `apiRef.current.unstable_dataSourceCache` instead of wrapper methods?

Reply:
This is also a good alternative to my previous proposition of `getDataSourceCache`, I'll resolve the other conversation in favor of this one.